lib/models/user.ex

```diff
@@ -0,0 +1,43 @@
+defmodule SlackCoder.Models.User do
+  use SlackCoder.Web, :model
+
+  schema "users" do
+    field :slack, :string
+    field :github, :string
+    field :avatar_url, :string
+    field :html_url, :string
+    field :name, :string
+    field :monitors, StringList
+
+    embeds_one :config, SlackCoder.Models.User.Config
+
+    timestamps
+  end
+
+  @required_fields ~w(slack github config)
+  @optional_fields ~w(avatar_url html_url name config monitors)
+
+  @doc """
+  Creates a changeset based on the `model` and `params`.
+
+  If no params are provided, an invalid changeset is returned
+  with no validation performed.
+  """
+  def changeset(model, params \\ :empty) do
+    model
+    |> cast(params, @required_fields, @optional_fields)
+    |> unique_constraint(:slack)
+    |> unique_constraint(:github)
+  end
+
+  for field <- [:slack, :github] do
+    def unquote(:"by_#{field}")(name) when is_binary(name) do
+      from u in __MODULE__, where: u.unquote(field) == ^name
+    end
+    def unquote(:"by_#{field}")(names) when is_list(names) do
+      from u in __MODULE__, where: u.unquote(field) in ^names
+    end
+    def by_slack(name), do: to_string(name) |> unquote(:"by_#{field}")()
+  end
```

Comment:
@mtyeh411 Heres examples of simple metaprogramming. Because basically every "keyword" is actually a macro, and because we're inside a `do` block for the module body, we're effectively already inside a `quote do: ...` block. That means we can `unquote` to interpolate values in our code! Code interpolation FTW!